Tutorial.playground/Contents.swift

```diff
@@ -0,0 +1,658 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+// ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+// ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+// ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+// MARK: - Introduction
+
+//: The `Aquifer` library decouples stream processing stages from each other so
+//: that you can mix and match diverse stages to produce useful streaming
+//: programs.  If you are a library writer, `Aquifer` lets you package up
+//: streaming components into a reusable interface.  If you are an application
+//: writer, `Aquifer` lets you connect pre-made streaming components with minimal
+//: effort to produce a highly declarative program.
+//:
+//: To enforce loose coupling, components can only communicate using two
+//: commands:
+//:
+//: * `yield`: Send output data
+//:
+//: * `await`: Receive input data
+//:
+//: `Aquifer` has four types of components built around these two commands:
+//:
+//: * `Producer`s can only `yield` values and they model streaming sources
+//:
+//: * `Consumer`s can only `await` values and they model streaming sinks
+//:
+//: * `Pipe`s can both `yield` and `await` values and they model stream
+//: transformations
+//:
+//: * `Effect`s can neither `yield` nor `await` and they model non-streaming
+//: components
+//:
+//: You can connect these components together in four separate ways which
+//: parallel the four above types:
+//:
+//: * `for_` handles `yield`s
+//:
+//: * `>~` handles `await`s
+//:
+//: * `>->` handles both `yield`s and `await`s
+//:
+//: * `>>-` handles return values
+//:
+//: As you connect components their types will change to reflect inputs and
+//: outputs that you`ve fused away.  You know that you`re done connecting things
+//: when you get an `Effect`, meaning that you have handled all inputs and
+//: outputs.  You run this final `Effect` to begin streaming.
+
+import func Swiftz.const
+import func Swiftz.curry
+import func Swiftz.<|
+import Aquifer
+
+// MARK: - Producers
+
+//: `Producer`s are effectful streams of input.  Specifically, a `Producer` is a
+//: Type that extends any other type with a new `yield` command. This `yield` 
+//: command lets you send output downstream to an anonymous handler, decoupling
+//: how you generate values from how you consume them.
+
+//: As an aside: Swift does not *technically* allow for the definition of polymorphic
```

Comment:
Looks like we can only emphasize.  It may be possible to just stick this at the bottom and link to it.